app/models/package_manager/go.rb

```diff
@@ -245,7 +249,18 @@ def self.project_find_names(name)
           &.last
           &.sub(/https?:\/\//, "")
 
-        go_import&.start_with?(*KNOWN_HOSTS) ? [go_import] : [name]
+        chosen_name = go_import&.start_with?(*KNOWN_HOSTS) ? [go_import] : [name]
+        # Collect info on what data we're mapping here, to ensure it's doing the right thing.
+        StructuredLog.capture(
+          "GO_MODULES_PROJECT_FIND_NAMES",
+          {
+            name: name,
+            found_name: go_import,
+            chosen_name: chosen_name,
+          }
+        )
```

Comment:
I'm half-skeptical that this method is choosing the right name, so this log gives us some evidence towards that.